src/librustc_mir_build/build/mod.rs

```diff
@@ -77,22 +83,23 @@ fn mir_build(tcx: TyCtxt<'_>, def_id: DefId) -> BodyAndCache<'_> {
                     abi = Abi::Rust;
                     vec![ArgInfo(liberated_closure_env_ty(tcx, id, body_id), None, None, None)]
                 }
-                ty::Generator(..) => {
+                ty::Generator(def_id, substs, _) => {
                     let gen_ty = tcx.body_tables(body_id).node_type(id);
-                    vec![
-                        ArgInfo(gen_ty, None, None, None),
-                        ArgInfo(tcx.mk_unit(), None, None, None),
-                    ]
+                    let resume_ty = substs.as_generator().resume_ty(def_id, tcx);
```

Comment:
You should get the `resume_ty` from `gen_ty` as the `ty::Generator` above is supposed to have type parameters for `resume_ty`, `witness`, etc.

Reply:
Done, but I could also just use `tcx.mk_unit()` directly I guess. If you prefer that, I can switch to that.

Reply:
Why use `()` here? That doesn't seem to relate to the resume type at all.

Reply:
When the user doesn't provide an argument, the resume type is set to `()`. This code path only has to add an implicit argument if the user didn't write one, so it's always `()` in that case (unless I'm missing something).

Reply:
Ah yeah, you can just use `()` here.